src/commonReducers/clients/clientSlice.js

```diff
@@ -0,0 +1,46 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import sources from 'api/sources';
+
+
+export const fetchClientsById = createAsyncThunk(
+  'clients/fetchClientsStatus',
+  async (userID) => {
+    const response = await sources.clientList.get(userID);
+    return response.json();
+  },
+);
+
+export const addClient = createAsyncThunk(
+  'clients/getClients',
+  async (data) => {
+    const response = await sources.clientList.post(data);
+    return response;
+  },
+);
+
+export const clientsSlice = createSlice({
+  name: 'clients',
+  initialState: {
+    entities: [],
+    loading: false,
+    error: '',
+  },
+  reducers: {},
+  extraReducers: {
+    [fetchClientsById.pending]: (state) => {
+      state.loading = true;
+    },
+    [fetchClientsById.rejected]: (state, action) => {
+      state.error = action.payload;
+    },
+    [fetchClientsById.fulfilled]: (state, action) => {
+      state.loading = false;
```

Comment:
Our app is pretty simple now, but I wonder if this is clear enough - If I was working with this state somewhere else in the app, I would ask "what is loading here"? In this case it's all clients and their information.

so maybe this should be something like pendingGetClients or getClientsIsLoading?

Eventually you may want to have different loading states for different async calls is what I'm getting at. Like adding a new client for instance.